components/schema-migrator/migrations/director/20230801103000_add_trigger_for serv_sub_mappings_deletion.up.sql

```diff
@@ -0,0 +1,19 @@
+BEGIN;
+
+CREATE OR REPLACE FUNCTION delete_cert_subject_mapping_on_app_template_delete()
+RETURNS TRIGGER AS
+$$
+BEGIN
+    DELETE FROM cert_subject_mapping
+    WHERE internal_consumer_id = OLD.id::varchar(256);
+    RETURN OLD;
+END;
+$$
+LANGUAGE plpgsql;
+
+
+CREATE TRIGGER trigger_delete_cert_subject_mapping_on_app_template_delete
```

Comment:
Wouldnt it be better if the logic for deleting the certificate subject mappings is implemented in the service layer of the application template. If this approach is not suitable for the MAP operator another approach would be to list the certificate subject mappings for the consumer ID(new query could be introduced) and then delete the respective resources. Using APIs or implementing in the service layer will result in better traceability(we will have logs for the deleted resources) and will be easier to follow the logic of the flow. By using the trriger the business logic is kind of hidden and one should remember that there is a trigger for deleting the resource.

Reply:
It was implemented in business operation inside MAP operator, but we decide to move it lower in layers. 
Relation between CSM and AT is hard, and we should ensure all such hard links to be strictly kept. In short, Certificate Subject Mapping without Application template is a leak that cannot be traced and cleared. Such leak can happen in case business logic is not fully executed due to connectivity glitch, crash or bug in code. Having a trigger that handles this ensures such leaks will not occur. 
Similar is situation with all resources that are connected (have foreign keys).

Reply:
If the deletion of the CSM is processed in the service layer of the Application template the deletion of both of the resources will be performed in one db transaction. If any of the resources fails to be deleted the transaction will be rolled back. If that is not enough we can ensure that tere is no CSM resource left by deleting it before deleting the Application template

Reply:
I did not understand correctly the proposition. We will move the logic from Trigger to Service layer. No need of additional discussion.